integrasjonspunkt/src/main/java/no/difi/meldingsutveksling/nextbest/MessageOutController.java

```diff
@@ -0,0 +1,227 @@
+        return ResponseEntity.ok(conversationResource);
+    }
+
+    @RequestMapping(value = "/out/messages/{conversationId}", method = RequestMethod.POST)
+    @ResponseBody
+    public ResponseEntity uploadFiles(
+            @PathVariable("conversationId") String conversationId,
+            MultipartHttpServletRequest request) {
+
+        OutgoingConversationResource conversationResource = repo.findOne(conversationId);
+        if (conversationResource == null) {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).body("No conversation with supplied id found");
+        }
+
+        ArrayList<String> files = Lists.newArrayList(request.getFileNames());
+        for (String f : files) {
+            MultipartFile file = request.getFile(f);
+            log.info("Key: "+file.getName());
+            log.info("Name: "+file.getOriginalFilename());
+            log.info("Content-Type: "+file.getContentType());
+            log.info("Size (bytes): "+file.getSize());
+
+            String filedir = props.getNextbest().getFiledir();
+            if (!filedir.endsWith("/")) {
+                filedir = filedir+"/";
+            }
+            filedir = filedir+conversationId+"/";
+            File localFile = new File(filedir+file.getOriginalFilename());
+            localFile.getParentFile().mkdirs();
+
+            try {
+                FileOutputStream os = new FileOutputStream(localFile);
+                BufferedOutputStream bos = new BufferedOutputStream(os);
+                bos.write(file.getBytes());
+                bos.close();
+
+                if (!conversationResource.getFileRefs().values().contains(file.getOriginalFilename())) {
+                    conversationResource.addFileRef(file.getOriginalFilename());
+                }
+                repo.delete(conversationResource);
+            } catch (java.io.IOException e) {
+                log.error("Could not write file {f}", localFile, e);
+                return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body("Could not write file");
+            }
+        }
+
+        try {
+            messageSender.sendMessage(conversationResource);
+        } catch (MessageContextException | IOException e) {
+            log.error("Send message failed.", e);
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body("Error during sending. Check logs");
+        }
+
+        return ResponseEntity.ok().build();
+    }
+
+    @RequestMapping(value = "/tracings/{conversationId}", method = RequestMethod.GET)
+    @ResponseBody
+    public ResponseEntity getTracings(
+            @PathVariable("conversationId") String conversationId,
+            @RequestParam(value = "messagetypeId", required = false) String messagetypeId,
+            @RequestParam(value = "lastonly", required = false) boolean lastonly) {
+        List<Conversation> clist = convRepo.findByConversationId(conversationId);
+        if (clist == null || clist.size() == 0) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use isEmpty() to check whether the collection is empty or not. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS1155)